libs/brain/input-otp/src/lib/brn-input-otp.component.ts

```diff
@@ -0,0 +1,127 @@
+import { BooleanInput, NumberInput } from '@angular/cdk/coercion';
+import {
+	booleanAttribute,
+	Component,
+	computed,
+	forwardRef,
+	input,
+	model,
+	numberAttribute,
+	signal,
+} from '@angular/core';
+import { ControlValueAccessor, FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ChangeFn, TouchFn } from '@spartan-ng/brain/forms';
+import { provideBrnInputOtp } from './brn-input-otp.token';
+
+export const BRN_INPUT_OTP_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => BrnInputOtpComponent),
+	multi: true,
+};
+
+export type InputMode = 'text' | 'tel' | 'url' | 'email' | 'numeric' | 'decimal' | 'search';
+
+@Component({
+	selector: 'brn-input-otp',
+	standalone: true,
+	imports: [FormsModule],
+	template: `
+		<ng-content />
+		<div style="position: absolute; inset: 0; pointer-events: none;">
+			<input
+				[class]="inputClass()"
+				autocomplete="one-time-code"
+				data-slot="input-otp"
+				style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; textAlign: left; opacity: 1; color: transparent; pointerEvents: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: transparent solid 0px; box-shadow: none; line-height: 1; letter-spacing: -0.5em; font-family: monospace; font-variant-numeric: tabular-nums;"
```

Comment:
Do we need this styling at the brn level? I feel we could move this to hlm and not lose out on much, and we could recommend using the above `inputClass` input for this. Similarly, we could also expose an input for classes for the container div on Line 30 & host on Line 46.

What do you think?

Reply:
I think this is similar to the brn-radio styles on the input. They belong to the input, thus it is invisible but focusable in the ui. Input classes should be used for additional styles like for disabled state.

https://github.com/spartan-ng/spartan/blob/a3cd75d62a4b1729a588d5d373ce11a35420ae22/libs/brain/radio-group/src/lib/brn-radio.component.ts#L59

Reply:
Makes sense!